README.md

# joeysvowels

<!-- badges: start -->

<!-- badges: end -->

The `joeysvowels` package provides a handful of datasets, some subsets
of others, that contain formant measurements and other information about
the vowels in my own speech. The purpose of the package is to make vowel
data easily accessible for demonstrating code snippets when
demonstrating working with sociophonetic data.

There are no functions contained in `joeysvowels`; it’s a data-only
package.

## Installation

You can install `joeysvowels` through GitHub:

``` r
remotes::install_github("joeystanley/joeysvowels")
```

You can then load the package like normally:

``` r
library(joeysvowels)
```

## Contents

Currently, there are six datasets contained in `joeysvowels`:

  - `darla` is one that was prepared using pretty standard methods,
    using the DARLA web interface to automatically transcribe,
    force-align, and extract formants from the audio. The audio was me
    reading 300 prepared sentences. It’s a bit of a noisy dataset, so
    it’s a good example of working with real data. It is also very
    close to the format that FAVE exports its data, so any examples that
    use `darla` can be easily modified to another FAVE-produced
    spreadsheet.

  - `coronals` is a much cleaner, more controlled dataset. You can read
    about the methods by viewing the documentation (`?coronals`).
    Essentially, I read a bunch of (C)CVC(C) nonce words where the
    consonants were (almost) all coronal. All my vowel phonemes are
    represented. I aligned and extracted formants from the data myself.
    Four formants were extracted every 5% of the vowels’ durations, so
    great for demonstrating functions and visuals involving vowel
    trajectories.
    
      - `midpoints` is a subset of `coronals` and contains only the
        midpoints from F1 and F2.
    
      - `mouth` is a subset of `coronals` and is only the MOUTH (/aw/)
        vowel.
    
      - `mouth_lite` is a subset of `mouth` and trims away most of the
        columns and only contains 10 tokens.

  - `idahoans` contains formant measurements from 11 individuals from
    the state of Idaho in the US. This is great for testing
    normalization functions.

## Example

You can access the datasets using `data()`. I’ll briefly visualize the
datasets.

``` r
library(joeysvowels)
library(ggplot2)
library(tidyr)
library(dplyr)
```

The `darla` dataset is messy.

``` r
data(darla)
ggplot(darla, aes(F2, F1, color = vowel)) + 
  geom_point() + 
  scale_x_reverse() + 
  scale_y_reverse()
```

![](man/figures/darla%20plot-1.png)<!-- -->

The `coronals` trajectories show clear influence from the surrounding
consonants.

``` r
data(coronals)
avg_trajs <- coronals %>%
  group_by(vowel, percent) %>%
  summarize(across(c(F1, F2), mean)) %>%
  print()
```

    ## `summarise()` regrouping output by 'vowel' (override with `.groups` argument)

    ## # A tibble: 273 x 4
    ## # Groups:   vowel [13]
    ##    vowel percent    F1    F2
    ##    <fct>   <dbl> <dbl> <dbl>
    ##  1 LOT         0  454. 1616.
    ##  2 LOT         5  592. 1346.
    ##  3 LOT        10  648. 1283.
    ##  4 LOT        15  651. 1238.
    ##  5 LOT        20  661. 1178.
    ##  6 LOT        25  651. 1176.
    ##  7 LOT        30  636. 1152.
    ##  8 LOT        35  630. 1148.
    ##  9 LOT        40  633. 1153.
    ## 10 LOT        45  625. 1158.
    ## # … with 263 more rows

``` r
ggplot(avg_trajs, aes(F2, F1, color = vowel)) + 
  geom_path(aes(group = vowel), 
            arrow = arrow(angle = 20, length = unit(0.15, "in"), type = "closed")) + 
  scale_x_reverse() + 
  scale_y_reverse()
```

![](man/figures/coronals%20plot-1.png)<!-- -->

`midpoints` is pretty clean.

``` r
data(midpoints)
ggplot(midpoints, aes(F2, F1, color = vowel)) + 
  geom_point() + 
  scale_x_reverse() + 
  scale_y_reverse()
```

![](man/figures/midpoints%20plot-1.png)<!-- -->

`mouth` shows pretty good trajectories.

``` r
data(mouth)
ggplot(mouth, aes(percent, hz, color = formant)) + 
  geom_path(aes(group = traj_id))
```

![](man/figures/mouth%20plot-1.png)<!-- -->

`mouth_lite` is just smaller.

``` r
data(mouth_lite)
ggplot(mouth_lite, aes(percent, hz, color = formant)) + 
  geom_path(aes(group = traj_id))
```

![](man/figures/mouth_lite%20plot-1.png)<!-- -->

``` r
data(idahoans)
ggplot(idahoans, aes(F2, F1, color = vowel)) + 
  geom_point() + 
  scale_x_reverse() + 
  scale_y_reverse() + 
  facet_wrap(~speaker)
```

![](man/figures/idahoans%20plot-1.png)<!-- -->


NEWS.md
# joeysvowels 0.1.1

* Added the `idahoans` dataset.

# joeysvowels 0.1.0

* Created the package
* Added a `NEWS.md` file to track changes to the package.


R/mouth.R
#' Formant measurements from 81 tokens of the MOUTH vowel
#'
#' A dataset containing formant measurements from 81 tokens of the MOUTH (/aw/)
#' vowel. These came from generated nonce words. This dataset is relatively
#' clean and can be used for vowel formant example data.
#'
#' This dataset is a subset of the coronals dataframe. It only includes data
#' from the MOUTH vowel, from only F1 and F2, and reshaped into a "tall" format.
#' See \code{?coronals} for more information on how the dataset was prepared.
#'
#' @format A data frame with 2,758 rows and 12 variables:
#' \describe{
#'   \item{traj_id}{a unique identifier for each trajectory (that is, each combination of vowel_id and formant))}
#'   \item{vowel_id}{a unique identifier for each vowel token}
#'   \item{start}{the start time for that vowel}
#'   \item{end}{the end time for that vowel}
#'   \item{t}{the time where formants were extracted}
#'   \item{percent}{how far into the vowel's duration (in terms of percent of the duration) the formants were extracted. 0 = onset, 50 = midpoint, 100 = offset}
#'   \item{word}{the generated nonce word I read}
#'   \item{pre}{the consonant(s) before the vowel (if any)}
#'   \item{fol}{the consonant(s) after the vowel}
#'   \item{formant}{which formant did the data come from}
#'   \item{hz}{the formant measurements, in hz}
#' }
#'
"mouth"


R/coronals.R
#' Joey's pre- and post-coronal formants
#'
#' A dataset containing 14,424 formant measurements from 1,049 vowel
#' tokens. These came from generated nonce words. Formants are extracted at
#' 21 points along the duration of the vowel. This dataset is relatively clean
#' and can be used for vowel formant example data.
#'
#' I generated 333 nonce words of the form (C)CVC(C) with the following sounds:
#' \itemize{
#'    \item Onsets: /t/, /d/, /s/, /z/, n/, /h/, / /, /st/, /sn/
#'    \item Vowels: /i/, /ɪ/, /eɪ/, /ɛ/, /æ/, /a/, /ɔ/, /oʊ/, /ʊ/, /u/, /aɪ/, /aʊ/, /ɔɪ/
#'    \item Codas: /d/, /z/, /dz/.
#' }
#' Generally, only coronals were used. Clusters were okay. Codas were only
#' obstruents. All American English vowels were used. Not the most scientific
#' dataset, but sufficient for my purposes.
#'
#' Every combination of these levels was generated and repeated three times. The
#' resulting list was sorted randomly. I read them in a quiet environment, manually aligned
#' them, extracted formants using a Praat script (4 formants at 4500 Hz), and
#' filtered out the bad measurements.
#'
#' The result is a pretty clean dataset showing my vowel formant trajectories,
#' in the environment of a coronal consonant.
#'
#' The intended purpose of this data is so that I can quickly have a nice sample
#' at my disposal when illustrating R functions. However, you may use this
#' dataset however you please.
#'
#' Metadata about me:
#' White male, born in 1989 in suburban St. Louis where I lived  until I was
#' 18. Parents are from upstate New York and Minnesota. Lived in Utah, Brazil,
#' and Georgia as an adult. Data was recorded July 2020 (age 31).
#'
#' @format A data frame with 14,424 rows and 13 variables:
#' \describe{
#'   \item{vowel_id}{a unique identifier for each vowel token}
#'   \item{start}{the start time for that vowel}
#'   \item{end}{the end time for that vowel}
#'   \item{t}{the time where formants were extracted}
#'   \item{percent}{how far into the vowel's duration (in terms of percent of the duration) the formants were extracted. 0 = onset, 50 = midpoint, 100 = offset}
#'   \item{F1}{the F1 measurement}
#'   \item{F2}{the F2 measurement}
#'   \item{F3}{the F3 measurement}
#'   \item{F4}{the F4 measurement}
#'   \item{word}{the generated nonce word I read}
#'   \item{pre}{the consonant(s) before the vowel (if any)}
#'   \item{vowel}{the vowel class, in Wells' Lexical Sets}
#'   \item{fol}{the consonant(s) after the vowel}
#' }
#'
"coronals"


R/idahoans.R
#' Vowel formants from 11 Idahoans
#'
#' A dataset containing information on 1,100 vowel tokens produced by 10 people
#' from the state of Idaho in the United States. For each speaker, there are
#' ten tokens per canonical monophthong, randomly selected from a larger
#' dataset. Vowels are not preceding sonorants and not following coronal
#' consonants. For each token, F1, F2, F3, and F4 were extracted at the midpoint
#' of each vowel using a Praat script. The individuals consented to their data
#' being used in this way.
#'
#' This dataset is useful for testing and demonstrating vowel normalization
#' functions.
#'
#' @format A dataframe with 1,100 rows and 7 variables.
#' \describe{
#'   \item{speaker}{a unique identifier per speaker}
#'   \item{sex}{biological sex of the speakers}
#'   \item{vowel}{vowel, in ARPABET. This is a handy transcription system since
#'   all General American English vowels are represented using two-letter codes.}
#'   \item{F1, F2, F3, F4}{vowel formant measurements, in Hz}
#' }
"idahoans"


R/midpoints.R
#' Formant measurements from 564 vowel midpoints
#'
#' A dataset containing formant measurements from the midpoints of 564 vowel
#' tokens. These came from generated nonce words. This dataset is relatively
#' clean and can be used for vowel formant example data.
#'
#' This dataset is a subset of the coronals dataframe. It only includes data
#' from vowel midpoints. The diphthongs PRICE, MOUTH, and CHOICE are excluded.
#' See \code{?coronals} for more information on how the dataset was prepared.
#'
#' @format A data frame with 564 rows and 12 variables:
#' \describe{
#'   \item{vowel_id}{a unique identifier for each vowel token}
#'   \item{start}{the start time for that vowel}
#'   \item{end}{the end time for that vowel}
#'   \item{t}{the time where formants were extracted}
#'   \item{F1}{the F1 measurement}
#'   \item{F2}{the F2 measurement}
#'   \item{F3}{the F3 measurement}
#'   \item{F4}{the F4 measurement}
#'   \item{word}{the generated nonce word I read}
#'   \item{pre}{the consonant(s) before the vowel (if any)}
#'   \item{vowel}{the vowel class, in Wells' Lexical Sets}
#'   \item{fol}{the consonant(s) after the vowel}
#' }
#'
"midpoints"


R/mouth_lite.R
#' Formant measurements from 10 tokens of the MOUTH vowel
#'
#' This is a "lite" version of the \code{mouth} dataset.
#'
#' This dataset contains only five columns and and only 10 random tokens of the
#' original \code{mouth} dataset. See \code{?mouth} for more
#' information. Ultimately, this is a subset of the \code{coronals} dataset, so
#' please see \code{?coronals} for information on how the data was created.
#'
#' @format A data frame with 352 rows and 5 variables:
#' \describe{
#'   \item{traj_id}{a unique identifier for each trajectory}
#'   \item{word}{the generated nonce word I read}
#'   \item{percent}{how far into the vowel's duration (in terms of percent of the duration) the formants were extracted. 0 = onset, 50 = midpoint, 100 = offset}
#'   \item{formant}{which formant did the data come from}
#'   \item{hz}{the formant measurements, in hz}
#' }
#'
"mouth_lite"


R/darla.R
#' Vowel formants produced by DARLA and FAVE
#'
#' A dataset containing information on 3,504 vowel tokens. It is formatted
#' very nearly how FAVE outputs its spreadsheets, so it is useful for
#' demonstrating code assuming you have a FAVE-produced spreadsheet.
#'
#' To create this dataset, I first selected 300 sentences from COCA. This was
#' part of a research project and I wanted to run through the sentences I would
#' have participants read by going through them myself. The sentences were
#' chosen because they contained key words I wanted to elicit.
#'
#' It is not the cleanest dataset. I sat at my kitchen table and read the 300
#' sentences with a decent microphone, but I had a bit of sore throat, so my
#' voice was not how it normally sounded. Plus, there was an infact screaming
#' during part of the recording. For my purposes, I didn't need a particularly
#' clean dataset, so I was fine with that. This was sometime around June 2017.
#'
#' I processed the data using DARLA (\url{http://darla.dartmouth.edu}) using
#' its automatic speech recognition option. I didn't have the time to transcribe
#' the data myself, and I didn't find it necessary, really. DARLA admits that
#' its word-level transcriptions are not completely reliable, but that it does
#' do a pretty good job at at least getting the vowels correct. Comparing this
#' output to the script I read, I'd say that's about right.
#'
#' After DARLA did its automatic transcription, it sent the audio and
#' transcription to be force aligned. At the time, Prosody-Lab was used.
#'
#' Finally, DARLA sends the audio and phoneme-level transcriptions to FAVE for
#' formant extraction. The resulting dataset is very nearly the same as the one
#' that FAVE produced. The only modifications are in the first few metadata
#' columns.
#'
#' This dataset is useful for two reasons. First, I use this code in a lot of my
#' R workshops; if you have FAVE-produced data yourself, you'll be able to
#' follow along using your own data with minimal modifications to the code.
#' Second, it's a great example of noisy data, which is useful when
#' demonstrating functions that detect outliers.
#'
#' Metadata about the speaker:
#' White male, born in 1989 in suburban St. Louis where I lived  until I was
#' 18. Parents are from upstate New York and Minnesota. Lived in Utah, Brazil,
#' and Georgia as an adult. Data was recorded July 2020 (age 31).
#'
#' @format A dataframe with 3,504 rows and 43 variables. There are probably
#' better explanations of these variables elsewhere on the internet, but here's
#' my description:
#' \describe{
#'   \item{name}{speaker name. Since this is just my data, it's just "joey"}
#'   \item{sex}{speaker sex. Since it's me, it's "M" for "male"}
#'   \item{vowel}{vowel, in ARPABET. Since DARLA uses the
#'   \href{http://www.speech.cs.cmu.edu/cgi-bin/cmudict}{CMU Pronouncing
#'   Dictionary}, it uses ARPABET for transcription of the vowels. This is a
#'   handy transcription system since all General American English vowels are
#'   represented using two-letter codes.}
#'   \item{Other columns will be added later.}
#' }
"darla"


# vowel = the vowel phoneme, transcribed in ARPABET.
# stress = primary (1), secondary (2), or unstressed (0).
# pre_word = the previous word
# word = the word that this vowel token is taken from
# fol_word = the following word
#
# Basic acoustic measurements
# F1, F2, F3 = formant measurements. They are taken at different time points along the duration of the vowel (some are one-third of the way into the duration, others are at the onset). See page. 37–38 of the attached article for details.
# F1_LobanovNored_unscaled, F2_LobanovNormed_unscaled = normalized versions of the F1 and F2 columns. Normalization is where you adjust the formant values on a by-speaker basis to remove physiological differences while also retaining sociolinguistic differences. The Lobanov method is ones such technique. (There is some debate as to whether it should be used or not, but that’s a whole nother rabbit hole I won’t take you down…)
# B1, B2, B3 = bandwidths for F1, F2, F3, measured in Hz. I’m a little foggy as to what these mean, but basically the wider the bandwidth, the less certain you are about the formants. Think of looking at a spectrogram in Praat: if the formants are very thick and black, you can’t really tell where the “true” measurements are. But if they’re narrow, you’ve got a better guess.
# t = the time (in seconds) into the recording that the F1, F2, F3 measurements were taken.
# beg, end = the beginning and end times (in seconds) of the vowel
# dur = the duration of the vowel (in seconds)
#
# Context columns. The next several columns all start with “plt” which stands for “Plotnik” a software developed and used by UPenn Linguistics that is probably now outdated (thanks, R). These are codes that have to do with the phonological categories of the previous and following segments. Makes it easy to sort by “all vowels before nasals” for example.
# plt_vlass = a different way of transcribing the vowels. These codes have some more nuance to them, i.e. “eyF” is (I believe) /e/ in an open (=“free”) syllable while “eyC” is /e/ in a closed (=“checked”) syllable. See this blog post by Joe Fruehwald (the maintainer of FAVE) for details.
# plt_manner = the manner of articulation for the following consonant
# plt_place = the place of the following consonant. I think they’re mostly straightforward, but I remember that “central” basically means /r/.
# plt_voice = the voicing of the following consonant
# plt_preseg, plt_folseg = something to do with the previous and following segments. I don’t know exactly. I’ve never used these columns but they’re part of the output we get so we keep them.
# pre_seg, fol_seg = the ARPABET transcription of the previous and following segments.
# context = basically, where in the word does the vowel appear.
# vowel_index = I guess these are numbers for each vowel phoneme? Like 1 = /æ/, 2 = /aɪ/, etc. Probably something that Plotnik uses.
# pre_word_trans, word_trans, fol_word_trans = full transcriptions of the previous, current, and following words, in ARPABET.
#
# Formant Measurements. This is really where the good stuff is. In addition to formant measurements at the determined points as outlined in the attached article, the program we use (FAVE) also takes points at 5 places along the vowel’s duration: 20% into the duration of the vowel, 35%, 50% (the midpoint), 65%, and 80%. Combined, these will give you decent information about the trajectory of the vowel if you want to go that route. If not, most people just use the midpoints. I think originally the columns were called something like “F1@20%” meaning “F1 measurement at 20% into the duration of the vowel”, but R has trouble with column names with “%” and “@“ so it converted them to “F1.20.” which is super annoying. We have unnormalized and normalized versions of each.
# F1.20., F2.20., F1.35., F2.35., F1.50., F2.50., F1.65., F2.65., F1.80., F2.80. = formant measurements.
# Same as above, with with _lob attached to the end = Lobanov-transformed versions of the previous 10 columns. Out of all the formant measurement columns this spreadsheet has, my guess is that the two you’ll want are F1.50._lob and F2.50._lob, which are normalized F1 and F2 at the midpoints. That’ll make them most comparable between speakers and midpoints are the most common thing to look at.
# Bark_height, Bark_backness = normalized versions of the midpoints (I believe) but transformed into the Bark scale.
# nFormants = a number between 3 and 6 that isn’t useful for most people but is used for diagnosing problems with formant measurements if we want to go back and pop the hood on FAVE.
#
# Metadata
# The filename where the data came from.
# State, sector, sector_name, sex, ethnicity, age, age_level, education_level, social_class, classification, kurath_type, town, county, land_region, locality, interview_year, birth_year, = metadata about the people that is mostly self-explanatory. I give some detail in the GSV if you turn on the help (the upper right corner on the plotting pages.)}
